functions/Get-DbaServerRole.ps1

```diff
@@ -0,0 +1,139 @@
+
+		.PARAMETER EnableException
+			By default, when something goes wrong we try to catch it, interpret it and give you a friendly warning message. This avoids overwhelming you with "sea of red" exceptions, but is inconvenient because it basically disables advanced scripting. Using this switch turns this "nice by default" feature off and enables you to catch exceptions with your own try/catch.
+
+		.NOTES
+			Tags: ServerRole, Security
+			Original Author: Shawn Melton (@wsmelton)
+
+			Website: https: //dbatools.io
+			Copyright: (C) Chrissy LeMaire, clemaire@gmail.com
+			License: GNU GPL v3 https: //opensource.org/licenses/GPL-3.0
+
+		.LINK
+			https://dbatools.io/Get-DbaServerRole
+
+		.EXAMPLE
+			Get-DbaServerRole -SqlInstance sql2016a
+
+			Outputs list of server-level roles and logins that are members for sql2016a instance.
+
+		.EXAMPLE
+			Get-DbaServerRole -SqlInstance sql2016a -Role sysadmin
+
+			Outputs members of sysadmin server-level role on sql2016a instance.
+
+		.EXAMPLE
+			Get-DbaServerRole -SqlInstance sql2017a -ExcludeFixedRole
+
+			Outputs the user server-level role(s) on sql2017a instance. A SQL Server 2017 supported feature only.
+
+		.EXAMPLE
+			Get-DbaServerRole -SqlInstance sql2016a -Login Bob
+
+			Outputs the server-level role(s) that the login Bob is a member of on sql2016a instance.
+
+			If Bob is not a member of any role, no output is returned.
+
+		.EXAMPLE
+			Get-DbaServerRole -SqlInstance sql2016a -Role sysadmin -Login Bob
+
+			Outputs the server-level role sysadmin with the login Bob as a member on sql2016a instance.
+
+			If Bob is not a member of that role, no output is returned.
+	#>
+	[CmdletBinding()]
+	param (
+		[Parameter(Position= 0, Mandatory= $true, ValueFromPipeline= $true)]
+		[DbaInstance[]]$SqlInstance,
+		[PSCredential]$SqlCredential,
+		[object[]]$ServerRole,
+		[object[]]$ExcludeServerRole,
+		[switch]$ExcludeFixedRole,
+		[Parameter(ValueFromPipeline=$true, ValueFromPipelineByPropertyName=$true)]
+		[object[]]$Login,
+		[switch]$EnableException
+	)
+
+	process {
+		foreach ($instance in $SqlInstance) {
+			try {
+				Write-Message -Level Verbose -Message "Connecting to $instance"
+				$server = Connect-SqlInstance -SqlInstance $instance -SqlCredential $SqlCredential
+			}
+			catch {
+				Stop-Function -Message "Failure" -Category ConnectionError -ErrorRecord $_ -Target $instance -Continue
+			}
+
+			$roles = $server.Roles
+
+			if ($ServerRole) {
+				$roles = $roles | Where-Object Name -In $ServerRole
+			}
+			if ($ExcludeServerRole) {
+				$roles = $roles | Where-Object Name -NotIn $ExcludeServerRole
+			}
+			if ($ExcludeFixedRole) {
+				$roles = $roles | Where-Object IsFixedRole -eq $false
+			}
+
+			foreach ($role in $roles) {
+				$members = $role.EnumMemberNames()
+				if ($Login) {
+					$members = $members | Where-Object {$_ -in $Login}
+				}
+
+				if ($members) {
+					foreach ($member in $members) {
```

Comment:
??
if the role has 7 members, you add ComputerName, InstanceName, SqlInstance and Login 7 times?

Reply:
Correct. It will generate 7 objects.

Reply:
Screen shot shows what this looks like for sysadmin role.

Reply:
That definitely belongs in another function:
from `Get-DbaServerRole` I expect 1 object for each role.
This output is `RoleMember`

Reply:
Even though your output of role member function will not be an SMO object? Why would we need an extra function to just output strings?

Reply:
Enumerate for members of server level roles is part of the Roles object. It is not a seperate object in SMO. So not sure I follow the logic of making it a seperate function.

Reply:
As well, how do you see use of two commands? It would be extra overhead to pipe the same object to pull the members when they are already in the role object. I have no reason to call a RoleMember function when I can just as easily call the enumerate method that is returned from the server role command.

Reply:
How about adding it as a collection property?

![image](https://user-images.githubusercontent.com/11204251/33000952-9dfe82f0-cd70-11e7-9abc-22eba66633d1.png)

Reply:
Just adding the Login as a collect equates to doing the same thing with the `EnumMemberNames()`

![image](https://user-images.githubusercontent.com/11204251/33001059-3c90db0c-cd71-11e7-82d5-af32b11acf58.png)

Reply:
that is acceptable, 
although this is also possible without adding the property:
`$r.EnumServerRoleMembers()` is already available.

Reply:
The motivation for having `Get-DbaRoleMember` as a separate function is
1. It's not for you and me, but for DBAs who don't know how to enumerate members and consequently handle that output.
2. It's not correct to return members as multiplied Role objects.

We certainly need to examine the `RoleMember` function(s). The existing one can be improved, no doubt. We have to decide whether we combine or split the Database and Server Role members. Also if this could include group memberships and so on. Functionality that can make it more useful, even for you and me.
But that's for later.